corehq/apps/cloudcare/static/cloudcare/js/util.js

```diff
@@ -210,9 +210,20 @@ hqDefine('cloudcare/js/util',['hqwebapp/js/initial_page_data', 'integration/js/h
         };
     }
 
+    var addDelegatedClickDispatch = function (linkTarget, linkDestination) {
+        document.addEventListener('click', function (event) {
+            if (event.target.target === linkTarget) {
+                linkDestination(event.target);
+                event.preventDefault();
+            }
+        }, true);
+    };
+
     var injectMarkdownAnchorTransforms = function () {
         if (window.mdAnchorRender) {
+            var HMACCallout = hqImport('integration/js/hmac_callout');
```

Comment:
@orangejenny I tried adding the HMACCallout to the module includes

https://github.com/dimagi/commcare-hq/pull/28422/files#diff-ae5101e18c1cf43bcb1e769c60dac202R3

but I seem to recall you mentioning that cloudcare / formplayer doesn't actually use RequireJS, which means I need to import it here explicitly, right?

I'm getting a test failure when using this pattern, though, so not sure if this is on the right track.

Reply:
Short answer: add `integration/js/hmac_callout` to the dependency list at the top of of this file, even though yes, that is bad practice. And also please add `jquery` to the list and `$` to the params.

Long answer:

Best practice is not to include an explicit dependency list for modules that haven't been tested on requirejs, because if you miss something, it'll introduce a bug that won't surface until the module *is* used on a requirejs page, and tracking down those bugs is terrible. 

However, I asked you to add this file's dependency list because I was certain it was correct (even though I was wrong, it's missing jquery), and I wanted to avoid having to think much about the HMAC code when web apps gets migrated to RequireJS. But yeah, it's now breaking a test because if cloudcare *were* using requirejs, the `hqImport` would be a problem.

I just added docs about this earlier this week: https://github.com/dimagi/commcare-hq/blob/master/docs/js-guide/migrating.md#note-on-editing-unmigrated-javascript

Migrating web apps to RequireJS is getting higher on my priority list every day.

Reply:
@orangejenny I'm still a bit stuck on this, unfortunately.

I updated the file header as per your suggestion
https://github.com/dimagi/commcare-hq/pull/28422/files#diff-ae5101e18c1cf43bcb1e769c60dac202R3

but the test runner still doesn't like the hqImport statement. Since the setup isn't actually in RequireJS, though, I can't remove it since the HMACCallout isn't actually defined by the dependency definition.

Reply:
Hmm, that's aggravating. I'll take a look.

Reply:
In `dependencies.html`, move the `integration/js/hmac_callout.js` script tag to be above the one for `cloudcare/js/util.js`.

On non-RequireJS pages, [hqModules](https://github.com/dimagi/commcare-hq/blob/92656804df7513dc95826572d7c270855fbf293f/corehq/apps/hqwebapp/static/hqwebapp/js/hqModules.js#L75-L82) does reads the dependencies specified in the `hqDefine` and attempt to populate the corresponding parameters, but since it can't actually load modules, it only has access to scripts that have already been added to the page. So the ordering of script tags becomes important (and fragile).

Reply:
Aha, that explains why things behave in slightly different ways in different files, thanks for clarifying. Have updated to match the pattern and should have a much clearer time doing this moving forward, thanks.